flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -235,9 +242,8 @@ public void testUpdateChannelBeforeRequest() throws Exception {
 			.buildUnknownAndSetToGate(inputGate);
 
 		// Update to a local channel and verify that no request is triggered
-		inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(
-			unknown.partitionId,
-			ResultPartitionLocation.createLocal()));
+		ResourceID location = ResourceID.generate();
+		inputGate.updateInputChannel(location, createPartitionInfo(unknown.getPartitionId(), location));
```

Comment:
ditto: use default `ResourceID` in `NettyShuffleDescriptorBuilder`

Reply:
I think the random one provides a better check for tests.